CMSgov/easi-app
# Fix configuration errors that prevented golangci-lint from properly r…

# NOREF

## Description

This PR fixes some configurations in our `.pre-commit-config.yaml` and `.golangci.yml` files. These fixes are:

- `.pre-commit-config.yaml`
  - Running `golangci-lint-full` instead of `golangci-lint` (the latter only ever lints added files in Git -- the former always scans the whole repo). This _does_ mean each run of the linter might take longer locally, but it's a bit more bulletproof.
- `.golangci-lint.yaml`
  - Set `exclude-generated` to `disabled`. This is because we often edit files that linters consider to be generated (i.e. anything containing a comment along the lines of `// DO NOT EDIT` or `// Code generated by ...`. The most notable instance of this is [schema.resolvers.go](https://github.com/CMS-Enterprise/easi-app/blob/23db8ba49a940fa24e83dcfd1b6e678bcf5abe3b/pkg/graph/schema.resolvers.go#L5).
    - As part of this solution, I _did_ try modifying the config for gqlgen to exclude such comments, but since `generated.go` is where all the resolver types live (i.e. `businessCaseResolver`), the linter sees our `schema.resolvers.go` as something that's called _by_ a generated file, and considers it generated as well.
    - To account for this change, I added some `exclude-files` to manually exclude generated files that we _actually_ don't want to touch. If any other files accidentally get linted, we should add them to this list!

## How to test this change

A neat way to validate that this works is to find a `switch` statement in our app, remove the `default` case (and maybe another case) and see if your editor (if configured to use golangci-lint -- [here's how to do so with VSCode](https://golangci-lint.run/welcome/integrations/)!) warns you. Otherwise, running `scripts/dev lint` should catch it!

## PR Author Checklist




## PR Reviewer Guidelines

- It's best to pull the branch locally and test it, rather than just looking at the code online!
- When approving a PR, provide a reason _why_ you're approving it
  - e.g. "Approving because I tested it locally and all functionality works as expected"
  - e.g. "Approving because the change is simple and matches the Figma design"
- Don't be afraid to leave comments or ask questions, especially if you don't understand why something was done! (This is often a great time to suggest code comments or documentation updates)
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
